### PR TITLE
BR-154 Add Docker Credentials to prevent exceeding the Docker call limits

### PR DESCRIPTION
**Pull request by TonyMorrison**
#### What does this PR do?
Added Docker credentials to maven steps that do anonymous calls to Docker Hub to prevent exceeding the Docker call limits.
Docker recently instituted new limits for anonymous pulls from the Docker Hub registry, and by using the bot with the credential to access, we effectively double the number of calls we can make vs the anonymous calls

#### Who is reviewing it? 
@jlcsmith 
@LinkMJB 

#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)

@jlcsmith
@shaundmorris

#### What are the relevant tickets?

[CAL-xxx](https://codice.atlassian.net/browse/CAL-xxx)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Review by shaundmorris — APPROVED** [review]

**Comment by TonyMorrison** [comment]
build now

**Review by jlcsmith — APPROVED** [review]

**Comment by TonyMorrison** [comment]
build now

File: Jenkinsfile
Comment (LinkMJB) [inline Jenkinsfile:38]: Where is this variable being used?

File: Jenkinsfile
Comment (LinkMJB) [inline Jenkinsfile:83]: For some reason, these Pipeline/Jenkinsfile changes do not seem to have taken effect in the PR build...

Screen Shot 2021-01-26 at 12 11 03 PM

**Review by LinkMJB — CHANGES_REQUESTED** [review]

File: Jenkinsfile
Comment (TonyMorrison) [reply Jenkinsfile:38]: see DOCKERHUB_CREDS_USR and DOCKERHUB_CREDS_PSW, are automatically generated from that variable
https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#usernames-and-passwords

File: Jenkinsfile
Comment (TonyMorrison) [reply Jenkinsfile:83]: Cloudbees insinuates that to change the Jenkinsfile is to change the job configuration, and therefore the user initiating the PR must be trusted or else the PR will revert to the trusted configuration previously used. Seems weird that I am the PR guy here and I am an admin and it still doesnt trust me.. :-s
https://support.cloudbees.com/hc/en-us/articles/227246367-Why-Jenkinsfile-changes-Are-Not-Reflected-in-PR-Build#resolution

File: Jenkinsfile
Comment (LinkMJB) [reply Jenkinsfile:38]: Wow, I did not know about this! Thanks for explaining it to me @TonyMorrison!

File: Jenkinsfile
Comment (LinkMJB) [reply Jenkinsfile:83]: I think this might be a bug with the version of Jenkins and the plugins we are using, identified here: https://issues.jenkins.io/browse/JENKINS-54126?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel -- I'm not going to hold up your PR from getting merged, just because of a problem with our Jenkins, but calling it out here just in case builds suddenly start failing after merging this. Hopefully these pipeline changes are respected after merging!

**Review by LinkMJB — APPROVED** [review]

**Pull request merged**
